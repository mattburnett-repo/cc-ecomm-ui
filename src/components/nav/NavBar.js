export default function NavBar () {
    return (
        <div>
            <div role="presentation" aria-label="header-message">header message</div>
            <button aria-label="logout">Log Out</button>
        </div>
    )
}