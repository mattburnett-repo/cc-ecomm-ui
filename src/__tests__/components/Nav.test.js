// 20211217: any wireframe will do...

import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

import NavBar from '../../components/nav/NavBar';

describe('<NavBar /> component tests', () => {
    beforeEach(() => render(<NavBar />));

    it('should display a header message', () => {
        screen.getByRole('presentation', {name: /header-message/i});
    });
    it('should display a home button if user is not on landing page', () => {
        expect(true).toBeFalsy();
    });
    it('should display a show cart button if user is not on landing page', () => {
        expect(true).toBeFalsy();
    });
    it('should display a checkout button if user is on the cart page', () => {
        expect(true).toBeFalsy();
    });
    it('should display a cancel button if user is on the shipping info or payment info pages', () => {
        expect(true).toBeFalsy();
    });
    it('should display a logout button', () => {
        screen.getByRole('button', {name: /logout/i});
    });

}); // end component

describe('<NavBar /> feature tests', () => {
    beforeEach(() => render(<NavBar />));

    it('doesnt do anything yet', () => {
        expect(true).toBeFalsy();
    });

    it('renders a snapshot', () => {
        const tree = renderer.create(<NavBar />).toJSON();
        expect(tree).toMatchSnapshot();
    });
}); // end feature

describe('<NavBar /> unit / functional tests', () => {
    beforeEach(() => render(<NavBar />));

    it('doesnt do anything yet', () => {
        expect(true).toBeFalsy();
    })
}); // end component