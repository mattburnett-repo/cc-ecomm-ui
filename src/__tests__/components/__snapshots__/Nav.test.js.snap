// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> feature tests renders a snapshot 1`] = `
<div>
  <div
    aria-label="header-message"
    role="presentation"
  >
    header message
  </div>
  <button
    aria-label="logout"
  >
    Log Out
  </button>
</div>
`;
